### PR TITLE
Various minor memory accounting fixes

### PR DESCRIPTION
**Pull request by nezihyigitbasi**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java [hunk 1]
```diff
@@ -165,7 +168,7 @@ private SortedPositionLinks(PositionLinks positionLinks, int[][] sortedPositionL
         this.positionLinks = requireNonNull(positionLinks, "positionLinks is null");
         this.sortedPositionLinks = requireNonNull(sortedPositionLinks, "sortedPositionLinks is null");
         this.lessThanFunction = requireNonNull(lessThanFunction, "lessThanFunction is null");
-        this.sizeInBytes = positionLinks.getSizeInBytes() + sizeOf(sortedPositionLinks);
+        this.sizeInBytes = INSTANCE_SIZE + positionLinks.getSizeInBytes() + (SizeOf.sizeOf(sortedPositionLinks[0]) * sortedPositionLinks.length);
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java]: Do all the arrays in sortedPositionLinks have the same length?

Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java]: I don't think so. I wanted to get a better approximation than what we have right now. If we want to get this totally accurate we need to traverse the array.

Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java]: Ok. Can you add a comment above that line clarifying that?

Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java]: Actually we can significantly overestimate with the previous version (e.g., int[1000][] and only the first few arrays are non-null and the rest are null). For now I traversed the array.

**Pull request merged**